src/phoebus_guibuilder/guibuilder.py

```diff
@@ -1,171 +1,129 @@
+            conf = yaml.safe_load(ioc)
+            entities = conf["entities"]
+            for entity in entities:
+                if (
+                    "P" in entity.keys() and entity["P"] == component_match
+                ):  # the suffix could be M, could be R
+                    self.valid_entities.append(
+                        Entry(
+                            type=entity["type"],
+                            DESC=None,
+                            P=entity["P"],
+                            M=None,
+                            R=None,
                         )
-                        return valid_entities
-                    else:
-                        print(f"No ioc.yaml file for service: {file}")
-
-
-def extract_valid_entities(ioc_yaml: str, component: Component) -> list[Entry]:
-    print(type(ioc_yaml))
-    entities: list[dict[str, str]] = []
-    valid_entities: list[Entry] = []
-    component_match = f"{component.P}:{component.R}"
-    with open(ioc_yaml) as ioc:
-        conf = yaml.safe_load(ioc)
-        entities = conf["entities"]
-        for entity in entities:
-            if (
-                "P" in entity.keys() and entity["P"] == component_match
-            ):  # the suffix could be M, could be R
-                valid_entities.append(
-                    Entry(type=entity["type"], DESC=None, P=entity["P"], M=None, R=None)
-                )
-
-    return valid_entities
-
-
-def gui_map(entrys: list[Entry]):
-    gui_map = "/dls/science/users/uns32131/BLGui/BLGuiApp/opi/bob/gui_map.yaml"
-
-    with open(gui_map) as map:
-        conf = yaml.safe_load(map)
-
-        for entry in entrys:
-            print(entry.type)
-            if conf[entry.type]:
-                print(
-                    conf[entry.type]["file"]
-                )  # Find correct .bob file, and injet macros
-                # TODO:  create a copy of the file, and replace the required macros
-                # TODO:  return the file to guibuilder
-
-            else:
-                print("No BOB available")
-
-
-# find_services_folders()
-# print(valid_entities)
-# gui_map(valid_entities)
+                    )
+
+    def gui_map(self, entrys: list[Entry]):
+        """
+        Maps the valid entities from the ioc.yaml file
+        to the required screen in gui_map.yaml
+        """
+
+        gui_map = "/BLGui/BLGuiApp/opi/bob/gui_map.yaml"
+
+        with open(gui_map) as map:
+            conf = yaml.safe_load(map)
+
+            for entry in entrys:
+                print(entry.type)
+                if conf[entry.type]:
+                    print(
+                        conf[entry.type]["file"]
+                    )  # Find correct .bob file, and injet macros
+                    # TODO:  create a copy of the file, and replace the required macros
+                    # TODO:  return the file to guibuilder
+
+                else:
+                    print("No BOB available")
+
+
+if __name__ == "__main__":
```

Comment:
we shouldn't need these two lines seeing as the command line should be handled by `__main__.py`; this script shouldn't need to be manually run.